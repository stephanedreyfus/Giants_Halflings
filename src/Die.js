import React from 'react';
import PropTypes from 'prop-types';
import { DieBtn } from './styling/DiceStyle';

export default function Die(props) {
  return(
    <DieBtn
      onClick={() => props.handleClick(props.idx)}>
      { props.val }
    </DieBtn>
  )
};

Die.propTypes = {
  locked: PropTypes.bool,
  handleClick: PropTypes.func.isRequired,
  idx: PropTypes.number,
  val: PropTypes.number,
};
