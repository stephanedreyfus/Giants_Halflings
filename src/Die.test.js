import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Die from './Die';

const handleClick = () => {};

it("renders without crashing", () => {
    shallow (<Die value="1" handleClick={handleClick}/>);
});

it("matches snapshot when unlocked", () => {
    const wrapper = shallow(<Die value="1" handleClick={handleClick}/>);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it("matches snapshot when locked", () => {
    const wrapper = shallow(<Die value="1" locked handleClick={handleClick}/>);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});