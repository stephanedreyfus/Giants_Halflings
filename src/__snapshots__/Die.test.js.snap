// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when locked 1`] = `
<styled.button
  locked={true}
  onClick={[Function]}
/>
`;

exports[`matches snapshot when unlocked 1`] = `
<styled.button
  onClick={[Function]}
/>
`;
